src/v/raft/heartbeat_manager.cc

```diff
@@ -149,7 +259,31 @@ heartbeat_manager::send_heartbeats(std::vector<node_heartbeat> reqs) {
           return ss::when_all_succeed(futures.begin(), futures.end());
       });
 }
+ss::future<>
+heartbeat_manager::send_heartbeats(std::vector<node_heartbeat_v2> reqs) {
+    return ss::do_with(
+      std::move(reqs), [this](std::vector<node_heartbeat_v2>& reqs) mutable {
+          std::vector<ss::future<>> futures;
+          futures.reserve(reqs.size());
+          for (auto& r : reqs) {
+              futures.push_back(do_heartbeat(std::move(r)));
+          }
+          return ss::when_all_succeed(futures.begin(), futures.end());
+      });
+}
+
+ss::future<> heartbeat_manager::do_dispatch_heartbeats_v2() {
```

Comment:
i also don't like the duplication here, it is unfortunate, i was a bit torn if i should make the code reusable for old and new heartbeats and came to the conclusion that in the end it would be harder to make it so. I will go over that again

Reply:
to be honest I am okay with it if generalizing these functions is too awkward

Reply:
or we can remove v1 heartbeats code in the next major release.